doc/companies.md

```diff
@@ -1 +1,980 @@
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "OK",
+    "data": {
+        "id":"5",
+        "company_code": "TC",
+        "name": "Test Company",
+        "address": "JL. Test Company",
+        "npwp": "123",
+        "nomor_kukuh": "123",
+        "tgl_kukuh": "12",
+        "phone_num": "021",
+        "fax_num": "021",
+        "seri_pajak": "12",
+        "kota": "JAKARTA"
+    }
+}
+```
+
+## <a name="edit"></a>Edit Data Companies
+
+### Endpoint
+PUT /companies/:id
+
+### Headers
+Key | Value
+--- | ---
+Content-Type | application/json
+Accept | application/json
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+id | string | 5
+company_code | string | TC  
+name | string | Test Company
+address | string | JL. Test Company
+npwp | string | 123
+nomor_kukuh | string | 123
+tgl_kukuh | date | 12
+phone_num | string | 021  
+fax_num | string | 021
+seri_pajak | string | 12
+kota | string | JAKARTA
+```
+{
+    "id":"5",
+    "company_code": "TC",
+    "name": "Test Company",
+    "address": "JL. Test Company",
+    "npwp": "123",
+    "nomor_kukuh": "123",
+    "tgl_kukuh": "12",
+    "phone_num": "021",
+    "fax_num": "021",
+    "seri_pajak": "12",
+    "kota": "JAKARTA"
+}
+```
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+404 | Not Found | User not found in database  
+500 | Internal Server Error | some un-handle error in server
+200 | OK | OK
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "OK",
+    "data": {
+        "id":"5",
+        "company_code": "TC",
+        "name": "Test Company",
+        "address": "JL. Test Company",
+        "npwp": "123",
+        "nomor_kukuh": "123",
+        "tgl_kukuh": "12",
+        "phone_num": "021",
+        "fax_num": "021",
+        "seri_pajak": "12",
+        "kota": "JAKARTA"
+    }
+}
+```
+
+### Logic
+
+#### Validation
+- id : required and not empty
+- company_code : required and not empty
+- name : required and not empty
+- address : required and not empty
+- npwp : required and not empty
+- phone_num : required and not empty
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Request Payload : empty
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "id is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+Request Payload :
+```
+{}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "id is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 3
+
+Request payload :
+```
+{
+    "id": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "id is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 4
```

Comment:
invalid case since we need no id. but we need the not found scenario by grab id from path.